presto-main/src/main/java/io/prestosql/sql/rewrite/ShowStatsRewrite.java

```diff
@@ -109,80 +94,44 @@ public Statement rewrite(
+            this.statsCalculator = requireNonNull(statsCalculator, "statsCalculator is null");
         }
 
         @Override
         protected Node visitShowStats(ShowStats node, Void context)
         {
             checkState(queryExplainer.isPresent(), "Query explainer must be provided for SHOW STATS SELECT");
 
-            Query query = getRelationQuery(node);
+            Query query = getRelation(node);
             QuerySpecification specification = (QuerySpecification) query.getQueryBody();
-
-            Plan plan;
-
-            try {
-                plan = queryExplainer.get().getLogicalPlan(session, query(specification), parameters, warningCollector);
-            }
-            catch (AccessDeniedException e) {
-                throw rewriteAccessDeniedException(e);
-            }
-
-            Table table = getTable(node, specification);
-            QualifiedObjectName tableName = createQualifiedObjectName(session, node, table.getName());
-            TableHandle tableHandle = metadata.getTableHandle(session, tableName)
-                    .orElseThrow(() -> semanticException(TABLE_NOT_FOUND, node, "Table '%s' not found", table.getName()));
-            TableMetadata tableMetadata = metadata.getTableMetadata(session, tableHandle);
-            Map<String, ColumnHandle> columnHandles = metadata.getColumnHandles(session, tableHandle);
-            Set<String> columnNames = extractStatsColumns(tableMetadata, specification.getSelect().getSelectItems());
-
-            try {
-                accessControl.checkCanSelectFromColumns(session.toSecurityContext(), tableName, columnNames);
-            }
-            catch (AccessDeniedException e) {
-                throw rewriteAccessDeniedException(e);
-            }
-
-            for (String columnName : columnNames) {
-                ColumnMetadata column = tableMetadata.getColumn(columnName);
-                if (!accessControl.getColumnMasks(session.toSecurityContext(), tableName, columnName, column.getType()).isEmpty()) {
-                    throw new PrestoException(NOT_SUPPORTED, "SHOW STATS for table with column masking is not supported: " + columnName);
-                }
-            }
-            if (!accessControl.getRowFilters(session.toSecurityContext(), tableName).isEmpty()) {
-                // SHOW STATS could reveal min, max value which would be otherwise filtered out; or could reveal statistical properties of the data, like row count
-                throw new PrestoException(NOT_SUPPORTED, "SHOW STATS is not supported for a table with row filtering");
-            }
-
-            validateShowStatsSubquery(node, query, specification, plan);
-
-            return rewriteShowStats(table, tableHandle, tableMetadata, columnNames, columnHandles, getConstraint(plan));
+            Plan plan = queryExplainer.get().getLogicalPlan(session, query(specification), parameters, warningCollector);
```

Comment:
`getRelation` creates artificial query (`simpleQuery(selectList(new AllColumns()), node.getRelation())`), which we here analyze and plan. This can lead to eg `AccessDeniedException` referencing ... things user did not write. Previously we would post-process this with `rewriteAccessDeniedException`. Why is this not needed anymore?

Reply:
@findepi Please see https://github.com/prestosql/presto/pull/3109#discussion_r524964177

Reply:
Thanks, let's talk about messages in https://github.com/prestosql/presto/pull/3109#discussion_r524964177
even if message is perfect, we should make sure we don't let some artificial node location to go through. But perhaps you do not create any artificial node locations, so this isn't a problem?

Reply:
> artificial node location

Artificial nodes do not get any location, so it should not be a problem.